ros_cross_compile/data_collector.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes to aid in the collection of benchmarking information."""
+
+import json
+
+
+class data_measurement:
+    """
+    Represents a measurement that has been collected.
+
+    Includes:
+    * name of measurement
+    * value of the measurement
+    * units of said value
+    """
+
+    def __init__(self, name: str, value: Any, unit: str, timestamp: str):
+        self.name = name
+        self.value = value
+        self.unit = unit
+        self.timestamp = timestamp
+
+    def __repr__(self):
+        return '[{}] {}: {} {}'.format(self.timestamp, self.name, self.value, self.unit)
+
+    def get_name(self):
```

Comment:
python doesn't need getters, you can delete all of these. unless you want to make the members read-only, in which case you should name them with underscore prefix e.g. `self._internal_value`, and define `@property` members to retrieve values. I'd say there's no reason for that right now though, you're fine without.

Reply:
You may even want to just make this a `collections.NamedTuple` if you want a simple read-only data structure